sungc/deglint.py

```diff
@@ -479,55 +580,61 @@ def nir_subtraction(self, vis_band_ids, nir_band_id, odir=None, waterVal=5):
         ------
         Exception
             * If any of the bands do not exist
+            * If any of the input data only contains nodata
         """
         # create output directory
         if not odir:
-            odir = pjoin(self.group_path, "DEGLINT", "GAO")
+            odir = self.group_path / "DEGLINT" / "MINUS_NIR"
 
-        if not os.path.exists(odir):
-            os.makedirs(odir, exist_ok=True)
+        if not odir.exists:
+            odir.mkdir(exist_ok=True)
 
         # Check that the input vis bands exist
-        self.check_bandNum_exist(self.band_ids, vis_band_ids)
-        self.check_bandNum_exist(self.band_ids, [nir_band_id])
+        self.check_bandnum_exist(self.band_ids, vis_band_ids)
+        self.check_bandnum_exist(self.band_ids, [nir_band_id])
 
         ix_nir = self.band_ids.index(nir_band_id)
-        nir_bandPath = str(self.bandList[ix_nir])
+        nir_bandpath = self.band_list[ix_nir]  # Path
 
         # ------------------------------ #
         #       load the NIR band        #
         # ------------------------------ #
-        with rasterio.open(nir_bandPath, "r") as nir_ds:
-            nir_nRows = nir_ds.height
-            nir_nCols = nir_ds.width
+        with rasterio.open(nir_bandpath, "r") as nir_ds:
+            nir_nrows = nir_ds.height
+            nir_ncols = nir_ds.width
             nir_im = nir_ds.read(1)
 
+            rio_funcs.check_image_singleval(nir_im, int(nir_ds.meta["nodata"]), "nir_im")
+
         # ------------------------------ #
-        nBands = len(vis_band_ids)
-        deglint_BandList = []
+        nbands = len(vis_band_ids)
+        deglint_bandlist = []
 
         # Iterate over each visible band
-        for z in range(0, nBands):
+        for z in range(0, nbands):
 
             ix_vis = self.band_ids.index(vis_band_ids[z])
-            vis_bandPath = str(self.bandList[ix_vis])
+            vis_bandpath = self.band_list[ix_vis]  # Path
 
             # ------------------------------ #
             #        load visible band       #
             # ------------------------------ #
-            with rasterio.open(vis_bandPath, "r") as dsVIS:
-                vis_im = dsVIS.read(1)
+            with rasterio.open(vis_bandpath, "r") as ds_vis:
+                vis_im = ds_vis.read(1)
 
                 # get metadata and load NIR array
-                kwargs = dsVIS.meta.copy()
+                kwargs = ds_vis.meta.copy()
                 nodata = kwargs["nodata"]
-                spatialRes = int(abs(kwargs["transform"].a))
+
+                rio_funcs.check_image_singleval(vis_im, int(nodata), "vis_im")
```

Comment:
```suggestion
                rio_funcs.check_image_singleval(vis_im, nodata, "vis_im")
```

Reply:
I'd not cast the nodata to int.  Your function supports `Union[float, int]`, could be expanded to `Union[float, int, None]`.
The case being _what if the type is None?_  You'll get an error forcing a None to an int.

Reply:
The issue is that nodata is typically a float, but the dtype of the bands are np.int16. If I don't change it to int() would the following still work?
`if np.all(image == value):`
where image is np.int16, while value is np.float64

Reply:
Yeah Python can compare an int array with a float value.

```python
>>> a = numpy.random.randint(0, 256, (3, 3)).astype('uint8')
>>> a
array([[ 25, 116,  60],
       [ 16, 195, 177],
       [195,  72,  88]], dtype=uint8)
>>> a == 25.0
array([[ True, False, False],
       [False, False, False],
       [False, False, False]])
```

The default return of any dataset opened by rasterio is `None` if there isn't a `nodata` field set.  So if `None` is the return nodata val, then I'd say skip the single value test.